src/main/java/io/stargate/sgv2/jsonapi/api/configuration/ObjectMapperConfiguration.java

```diff
@@ -43,24 +44,51 @@ private ObjectMapper createMapper(DocumentLimitsConfig documentLimitsConfig) {
             .enable(StreamReadFeature.USE_FAST_BIG_NUMBER_PARSER)
             .enable(StreamWriteFeature.USE_FAST_DOUBLE_WRITER)
             .build();
-    return JsonMapper.builder(jsonFactory)
-        // important for retaining number accuracy!
-        .enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)
+    JsonMapper mapper =
+        JsonMapper.builder(jsonFactory)
+            // important for retaining number accuracy!
+            .enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)
+
+            // case-insensitive enums, so "before" will match to "BEFORE" in an enum
+            .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
 
-        // case-insensitive enums, so "before" will match to "BEFORE" in an enum
-        .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+            // Verify uniqueness of JSON Object properties
+            .enable(StreamReadFeature.STRICT_DUPLICATE_DETECTION)
 
-        // Verify uniqueness of JSON Object properties
-        .enable(StreamReadFeature.STRICT_DUPLICATE_DETECTION)
+            /* 12-Dec-2023, tatu: Must not force use of Plain notation because
+             *  that caused: https://github.com/stargate/jsonapi/issues/726
+             *  where we can insert numbers that are not round-trippable
+             *  as they get expanded from more compact Engineering/Scientific
+             *  notation to longer Plain notation; effectively creating "unreadable"
+             *  Documents.
+             */
+            .disable(StreamWriteFeature.WRITE_BIGDECIMAL_AS_PLAIN)
+            .build();
 
-        /* 12-Dec-2023, tatu: Must not force use of Plain notation because
-         *  that caused: https://github.com/stargate/jsonapi/issues/726
-         *  where we can insert numbers that are not round-trippable
-         *  as they get expanded from more compact Engineering/Scientific
-         *  notation to longer Plain notation; effectively creating "unreadable"
-         *  Documents.
-         */
-        .disable(StreamWriteFeature.WRITE_BIGDECIMAL_AS_PLAIN)
-        .build();
+    mapper.addHandler(
+        new DeserializationProblemHandler() {
```

Comment:
I think it'd be good to extract custom `DeserializationProblemHandler` as static class here: that way tests could re-use it.